plugins/plugin-git/che-plugin-git-ext-git/src/main/java/org/eclipse/che/ide/ext/git/client/compare/selectablechangespanel/CheckBoxRender.java

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright (c) 2012-2017 Red Hat, Inc.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.che.ide.ext.git.client.compare.selectablechangespanel;
+
+import com.google.gwt.dom.client.Element;
+import com.google.gwt.dom.client.InputElement;
+import com.google.gwt.user.client.Event;
+import com.google.gwt.user.client.ui.CheckBox;
+import java.util.Comparator;
+import java.util.HashSet;
+import java.util.Set;
+import org.eclipse.che.ide.ext.git.client.compare.changespanel.ChangedFileNode;
+import org.eclipse.che.ide.ext.git.client.compare.changespanel.ChangedFolderNode;
+import org.eclipse.che.ide.resource.Path;
+import org.eclipse.che.ide.ui.smartTree.Tree;
+import org.eclipse.che.ide.ui.smartTree.TreeStyles;
+import org.eclipse.che.ide.ui.smartTree.data.Node;
+import org.eclipse.che.ide.ui.smartTree.presentation.DefaultPresentationRenderer;
+
+class CheckBoxRender extends DefaultPresentationRenderer<Node> {
+
+  private final SelectableChangesPanelView.ActionDelegate delegate;
+  private Set<Path> allNodePaths;
+
+  private final Set<Path> unselected;
+  private final Set<Path> indeterminate;
+
+  CheckBoxRender(TreeStyles treeStyles, SelectableChangesPanelView.ActionDelegate delegate) {
+    super(treeStyles);
+
+    this.delegate = delegate;
+    this.allNodePaths = new HashSet<>();
+    this.unselected = new HashSet<>();
+    this.indeterminate = new HashSet<>();
+  }
+
+  @Override
+  public Element render(
```

Comment:
Refactor the method. Split with smaller ones.

Reply:
done